src/VisualStudio/IntegrationTest/IntegrationTests/LanguageServerProtocol/LspGoToDefinition.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Common;
+using Roslyn.Test.Utilities;
+using Roslyn.VisualStudio.IntegrationTests.CSharp;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.LanguageServerProtocol
+{
+    /// <summary>
+    /// A set of LSP specific goto definition tests.
+    /// These tests test behavior that only applies to the LSP version of goto definition.
+    /// </summary>
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class LspGoToDefinition : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public LspGoToDefinition(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(LspGoToDefinition))
+        {
+        }
+
+        /// <summary>
+        /// This test corresponds to the <see cref="CSharpGoToDefinition.GoToDefinitionWithMultipleResults"/> test.
```

Comment:
What would this look like to have the non-LSP and LSP ones call into a single method that takes a boolean and just switches on what's different? That way as the tests evolve it's clear where the differences are, which should help in maintainability?

Reply:
Because AFAIK the only real difference is the different name of the window and having to call WaitForActiveWindow which dovetails into my complaint about the oddity that we're passing in an "expected name" to the GoToDefinition which is a no-op in this case.

Reply:
Done!  Changed to inherit from the original test class with a additional traits (setting up a static helper wouldn't be easy due to protected methods accessing instances)